ui/src/app/domains/views/DomainsList/DomainListHeader/DomainListHeader.tsx

```diff
@@ -13,20 +15,40 @@ const DomainListHeader = (): JSX.Element => {
   const domainCount = useSelector(domainSelectors.count);
   const domainsLoaded = useSelector(domainSelectors.loaded);
 
+  const [isFormOpen, setFormOpen] = useState(false);
```

Comment:
I don't think domains will need this, but there's also a `setSelected` pattern we use in the machine and KVM lists which also have generic types etc. This can be useful if you need to open the forms from outside the header:
https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/kvm/views/KVMDetails/KVMDetailsHeader/KVMDetailsHeader.tsx#L26

We can provide more details about how that's used if needed.